src/featurestorebundle/orchestration/PostActionsRunner.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+from logging import Logger
+from box import Box
+from featurestorebundle.orchestration.PostActionInterface import PostActionInterface
+from daipecore.decorator.ContainerManager import ContainerManager
+
+
+class PostActionsRunner:
+    def __init__(self, logger: Logger, post_actions: Box):
+        self.__logger = logger
+        self.__post_actions = post_actions
+
+    def run(self):
+        post_action_services = self.__get_post_action_services()
+
+        self.__validate_post_action_services(post_action_services)
+
+        if len(self.__post_actions) == 0:
+            self.__logger.info("No post actions")
+
+        for post_action_service, post_action in zip(post_action_services, self.__post_actions):
+            self.__logger.info(f"Running post action {post_action}")
+
+            post_action_service.run()
+
+            self.__logger.info(f"Post action {post_action} done")
+
+    def __get_post_action_services(self) -> List[PostActionInterface]:
+        post_actions_unified = [post_action[1:] if post_action.startswith("@") else post_action for post_action in self.__post_actions]
+
+        return [ContainerManager.get_container().get(post_action) for post_action in post_actions_unified]
+
+    def __validate_post_action_services(self, post_action_services: List[PostActionInterface]):
+        for post_action_service in post_action_services:
+            if not isinstance(post_action_service, PostActionInterface) or not hasattr(post_action_service, "run"):
```

Comment:
what if `run` isn't a method but a property?

Reply:
also doesn't every PostActionInterface have `run` from the parent?

Reply:
yes but it should raise not implemented if not implemented due to @abc decorator

Reply:
So basically the second condition is useless